staging/src/k8s.io/kube-aggregator/pkg/controllers/openapiv3/aggregator/aggregator.go

```diff
@@ -257,7 +264,16 @@ func (s *specProxier) handleGroupVersion(w http.ResponseWriter, r *http.Request)
 
 		for key := range apiServiceInfo.discovery.Paths {
 			if targetGV == key {
+				// The metrics logic is duplicated from InstrumentHandlerFunc in
+				// k8s.io/apiserver/pkg/endpoints/metrics/metrics.go
+				// We cannot use the function to wrap the handler here
+				// because a single OpenAPI handler can serve multiple
+				// group versions and should be recorded differently.
+				delegate := &metrics.ResponseWriterDelegator{ResponseWriter: w}
+				w = responsewriter.WrapForHTTP1Or2(delegate)
+
 				apiServiceInfo.handler.ServeHTTP(w, r)
+				metrics.MonitorRequest(r, "GET", "", "", "", "/openapi/v3/"+targetGV, "", "", false, "", delegate.Status(), delegate.ContentLength(), time.Since(requestReceivedTimestamp))
```

Comment:
It feels kinda janky to override subresource with GV especially when `MonitorRequest` actually takes `group` and `version` parameters independently.

Reply:
It's not a GV in the traditional sense, we have cases of independent paths that publish OpenAPI (eg:  /version, /logs, /.well-known/openid-configurations). Yeah I agree it's a bit janky, I just don't think it makes sense to use any of the other parameters.

Reply:
Right, don't put it in at all. There is no need to distinguish openapiv3 requests from each other in the metrics

Reply:
Yeah, I'm not quite sure what you're trying to measure. But if there isn't a need to disambiguate openapiv3 requests by GV, then I would prefer not doing this thing here.